### PR TITLE
Re-organize requirements files

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Re-organize requirements files to improve the file structure in the repository.

## How is this patch tested?

NA

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -147,8 +147,7 @@ If you plan on doing development and testing, you will also need to install the
 
 .. code-block:: bash
 
-    pip install -r doc-requirements.txt
-    pip install -r test-requirements.txt
+    pip install -r requirements/test-requirements.txt
```
Comment (liangz1) [inline CONTRIBUTING.rst:150]: Actually, I find the ordering here a bit annoying. When I read this doc, I follow the order to install `ip install -e .[extras]` first. After that, I read `If you plan on doing development and testing, you will also need to install the following into the conda environment`, where it asks you to install `requirements/test-requirements.txt` first and then `extras`. The result is, many libs are installed, uninstalled, and reinstalled back and forth (since some requirements have version limitations while others don't).

It would be convenient if you rephrase it to avoid developers installing `pip install -e .[extras]` first or make the requirements compatible (installing `test-requirements.txt` should not need to uninstall libs from `extras`).

**Review by liangz1 — APPROVED** [review]
LGTM. I agree with this re-org of the files. Because putting the requirements files under `dev` could confuse the developer to think that `dev` directory is used for setting up the personal development environment (which is actually used for github workflows).

Only one non-blocking comment about the wording.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (harupy) [reply CONTRIBUTING.rst:150]: Good point. Can ew just merge these code two blocks into one?

```bash
conda create --name mlflow-dev-env python=3.6
conda activate mlflow-dev-env
# installs mlflow from current checkout with packages required for development
pip install -e .[extras] -r requirements/test-requirements.txt
pip install -e tests/resources/mlflow-test-plugin
```

Comment (liangz1) [reply CONTRIBUTING.rst:150]: I'm not sure whether this will resolve the difficulty, since the audience might still install `[extras]` first and then have to uninstall the dependencies that do not meet `requirements/test-requirements.txt`.
Anyways, it just takes a longer time but is correct, so we don't have to spend too much time on it.
